src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -1144,6 +1158,13 @@ public boolean tripIsBanned(Trip trip) {
         return false;
     }
 
+    /**
+     * Check if trip route type matches for this plan.
+     */
+    public boolean tripRouteTypeMatches(Trip trip) {
+        return routeTypes == null || routeTypes.size() == 0 || routeTypes.contains(trip.getRoute().getType());
```

Comment:
I don't care for the double check.  Let "null" mean "anything is allowed, and don't worry about the empty set (your code above refuses to allow it to be set).